osu.Desktop/Program.cs

```diff
@@ -52,10 +52,6 @@ public static void Main(string[] args)
                 // See https://www.mongodb.com/docs/realm/sdk/dotnet/compatibility/
                 if (windowsVersion.Major < 6 || (windowsVersion.Major == 6 && windowsVersion.Minor <= 2))
                 {
-                    // If users running in compatibility mode becomes more of a common thing, we may want to provide better guidance or even consider
-                    // disabling it ourselves.
-                    // We could also better detect compatibility mode if required:
-                    // https://stackoverflow.com/questions/10744651/how-i-can-detect-if-my-application-is-running-under-compatibility-mode#comment58183249_10744730
```

Comment:
This comment can't just be removed. If compatibility options set are preventing the game from being launched *this code must detect it too and notify the user about it*. Because the notification added later *won't ever show if the game doesn't launch*.

Probably make `checkCompatibilityMode()` from `CompatibilityModeChecker` `public static` and call that here to handle that case.

Reply:
I think that we should rewrite the comment instead of keeping it. My idea is that once discussions are over, this PR will completely alleviate the issues we have with compatibility mode, therefore this comment is unnecessary (since we'd be replacing it with a description of our new implementation).

It would probably be good to hear from peppy if he would rather have lazer send a notification, or if he'd rather have lazer refuse to run with any compatibility mode flags set. We can get around to adapting this comment to our approach when we solidify which way we want to do this.

Reply:
> My idea is that once discussions are over, this PR will completely alleviate the issues we have with compatibility mode

No it won't if _compatibility mode prevents the game from running_.

The messagebox underneath this comment should be _checking_ if compatibility mode is enabled and only bringing it up if it detects it being enabled.

Reply:
I understand now. The code underneath the comment already disallows the game from running with compatibility mode. In this case, it would be wise to put the compatibility mode logic above the windows version checking logic. I'll fix this and add another comment with the diff to see if its satisfactory